core/src/banking_stage/qos_service.rs

```diff
@@ -208,11 +221,29 @@ impl QosService {
                 // Only transactions that the qos service included have to be
                 // checked for update
                 if let Ok(tx_cost) = tx_cost {
-                    match transaction_committed_details {
-                        CommitTransactionDetails::Committed { compute_units } => {
-                            cost_tracker.update_execution_cost(tx_cost, *compute_units)
-                        }
-                        CommitTransactionDetails::NotCommitted => cost_tracker.remove(tx_cost),
+                    if *transaction_committed_details == CommitTransactionDetails::NotCommitted {
+                        cost_tracker.remove(tx_cost)
+                    }
+                }
+            });
+    }
+
+    fn update_committed_transaction_costs<'a>(
+        transaction_cost_results: impl Iterator<Item = &'a transaction::Result<TransactionCost>>,
+        transaction_committed_status: &Vec<CommitTransactionDetails>,
+        bank: &Arc<Bank>,
+    ) {
+        let mut cost_tracker = bank.write_cost_tracker().unwrap();
```

Comment:
Not sure if the additional write locking and iteration (lines below) have noticeable performance impact. Can it be benchmarked?

Reply:
see https://github.com/solana-labs/solana/pull/32285#issuecomment-1610325957 (left top level for future)